src/components/share-url.ts

```diff
@@ -0,0 +1,42 @@
+import { LitElement, css, html } from "lit";
+import { customElement, state } from "lit/decorators.js";
+import { ICON_CHECK_SLOTTED, ICON_COPY_SLOTTED } from "../const";
+
+@customElement("share-url")
+export class ShareURL extends LitElement {
+  @state() private copied: boolean = false;
+
+  render() {
+    return html`
+      <md-text-button
+        trailing-icon
+        @click=${this.copyToClipboard}
+        ?disabled=${this.copied}
+      >
+        ${this.copied
+          ? html`URL copied to clipboard! ${ICON_CHECK_SLOTTED}`
+          : html`Copy URL to clipboard ${ICON_COPY_SLOTTED}`}
+      </md-text-button>
+    `;
+  }
+
+  async copyToClipboard() {
+    if (this.copied) return;
```

Comment:
I don't think this is necessary. Buttons that are disabled prevent all click events from happening.

Reply:
I put this here as the function was async so there might have been a chance of it calling twice if someone clicked fast enough. However, I agree, overkill solution